src/caduceus/outboundSender_test.go

```diff
@@ -0,0 +1,263 @@
+package main
```

Comment:
Only real comment I have to make on the tests is that you could use this https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks (subtests) to have a bit less redundant set up and tear down code if you find you're having issues with that.

For example if I want to test the `New()` and `Destroy()` methods on my `HelloWorld` object:
```
func TestHelloWorld(t *testing.T) {
    // <setup code>
    t.Run("TestHelloWorldNew", func(t *testing.T) { ... })
    t.Run("TestHelloWorldDestroy", func(t *testing.T) { ... })
    // <tear-down code>
}
```